integration_tests/test_load_connection.py

```diff
@@ -38,11 +39,13 @@ def on_market_cycle(self, market_info):
                 # Validate that the bid was deleted from the market
                 empty_listing = self.list_bids()
                 assert not any(b for b in empty_listing["bid_list"] if b["id"] == bid_info["id"])
+
                 # Place the bid with a rate that will be acceptable for trading
-                bid = self.bid_energy_rate(energy_requirement, 33)
+                bid = self.bid_energy_rate(energy_requirement, 33, replace_existing=False)
```

Comment:
Would be great to also test `replace_existing = True` in this or an additional integrationtest. 
Or do you think the unittests on the d3a are sufficient?

Reply:
I'm actually having a difficult time wrapping my head around d3a-api-client integration tests for some reasons. 

- I tried to test `replace_existing=True` by checking the number of bids available in `self.list_bids()`, but I have the feeling that the speed of the simulation is hindering this test. The list was almost always empty and I couldn't find an easy way to stop the process and debug it. Adding print statements to debug takes too long because of the running times, unfortunately.
- What the d3a-api-client integration tests are doing is running a simulation and applying a new strategy on it, using assertions to check that nothing breaks. In my opinion, this is not very stable nor scalable in the long term. I tried to create a new integration test and it would increase the running time of the test suite by 25 minutes, so I avoided it. The possibility of stopping a test after a few market slots isn't currently implemented I think. But most of all, the synchronization between the d3a simulation and the client requests can only be obtained by reducing the simulation speed, and it's not exactly a deterministic process. That's why I suggested we could try to implement a way to advance the simulation one slot at a time so that we could check its current state and make requests during that slot. We had a conversation on Slack a week ago where I suggested something like:
```python
d3_simulation.run_next_cycle()  # This could be a subprocess command for d3a
do_something()
assert something_about_t1  

d3_simulation.run_next_cycle()
do_something()
assert something_else_about_t2
```
I think @spyrostz wanted to create a new task on Jira for this, but I didn't check yet.

Reply:
@fievelk I generally agree, most of the time it is not needed that the simulation run a whole day (default duration=24h) for testing a single issue. How about changing the duration parameter in your "new" step function here: https://github.com/gridsingularity/d3a-api-client/pull/89/files#diff-bc44b0a5290e14efb9ff70fa0fbf4099cfda89eed3f5541781144eaa4bb070ea 
I also did something similar in my PR (and will adapt to your version once you pushed): https://github.com/gridsingularity/d3a-api-client/pull/97/files#diff-bc44b0a5290e14efb9ff70fa0fbf4099cfda89eed3f5541781144eaa4bb070ea
Afterwards we can go the proposed alternative way in the frame of the `TESTS-75` ticket.

For the empty bid_list I have s suspicion: They get traded right away. So you could suppress this by setting the price for the bid really low. This way these bids will stay in the market longer to be displayed in the list_bids. 

Let me know if you need more help.

Reply:
I tried changing the `--duration` parameter, but it had no effect. I think it is because of the two following chunks:

https://github.com/gridsingularity/d3a-api-client/blob/d9db6002378564f6d18d58ad9f1cbe9cfdcc0f20/integration_tests/steps/device.py#L92-L100

https://github.com/gridsingularity/d3a-api-client/blob/83c31965bb17f1a2033c608f129732c26ac545ab/integration_tests/test_load_connection.py#L59-L61

If I got it right, the `"finished"` flag will be set only when the market time reaches `23:00`. So I'm not sure the `duration` flag (and the actual duration of the simulation) is considered. But it seems odd, so I might be wrong.

Reply:
Ah yes, this is bad anyway, the `self.status = "finished" ` should only be changed when `on_finished` is triggered. So you need to add `on_finished` to the `AutoBidOnLoadDevice`  and `AutoBidOnPVDevice` and get rid of the last block.